.github/workflows/tests.yaml

```
@@ -29,7 +29,9 @@ jobs:
           cache: poetry
       - run: poetry install
       - name: Run tests
-        run: poetry run pytest tests/${{ matrix.test }}
+        run: poetry run pytest tests/${{ matrix.test }} --cov=tests/${{ matrix.test }} --cov-report=xml
+      - name: Upload coverage
+        uses: codecov/codecov-action@v2
```

Not sure it will make a huge difference but seems harmless:
```suggestion
        uses: codecov/codecov-action@v2
        if: always()
```